layers/+lang/emacs-lisp/packages.el

```diff
@@ -61,6 +61,25 @@
           "mhh" 'elisp-slime-nav-describe-elisp-thing-at-point)))))
 
 (defun emacs-lisp/init-emacs-lisp ()
+
+  (defun spacemacs//warn-on-old-byte-compiled-file ()
+    "Warn if it's emacs-lisp-mode and compiled file exists and is
+older than the current file."
+    (when (and (eq major-mode 'emacs-lisp-mode)
```

Comment:
A buffer-local `after-save-hook` is a better approach than checking the current mode.